Activate/simplified login experiment setup

hafizrahman opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #7614
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR adds the required wirings to let the A/B test experiment to be activated. It adds these functions:
- `activate()`: To start the testing
- `getCurrentVariant()`: To get the currently given variant.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

Nothing to test now since the Firebase A/B test is not running yet.

hafizrahman committed:
Add: Google Analytics tracking for activation and successful event.

hafizrahman committed:
Add: Internal analytics for simplified login.

hafizrahman committed:
Add: Simple function to get the current variant from Firebase.

Intentionally made as non-suspend function, as per this discussion https://github.com/woocommerce/woocommerce-android/pull/7617

hafizrahman committed:
Update: getCurrentVariant() to return actual value from Firebase.

hafizrahman committed:
Add: run() function that just starts the various analytics event.

- Track activation event in Google Analytics, which is needed by A/B test.
- Track also internally.

hafizrahman committed:
Fix: Detekt.

hafizrahman committed:
Update: Make code easier to follow. No functionality change.

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt:
```diff
@@ -90,6 +91,9 @@ class FirebaseRemoteConfigRepository @Inject constructor(
                 emit(JetpackTimeoutPolicyVariant.valueOf(defaultValues[JETPACK_TIMEOUT_POLICY_VARIANT_KEY]!!))
             }
 
+    override fun getSimplifiedLoginVariant(): String =
+        remoteConfig.getString(SIMPLIFIED_LOGIN_VARIANT_KEY)
```
Question: why not follow the same pattern as the other variants? I mean having the enums named same as the AB test values, which means for our case to either update the AB Test variants in Firebase Console or update the enum values.

And to be consistent with the rest, I think the conversion from String to enum should be part of the repository.

hafizrahman replied:
Ah, I did not notice that variant names in Firebase Console match the enums in other experiments. Good eye. +1 about conversing to enum in repository.

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/SimplifiedLoginExperiment.kt:
```diff
@@ -1,9 +1,42 @@
+    companion object {
+        // The two values below are set in Firebase
+        private const val VARIANT_CONTROL = "control"
+        private const val VARIANT_SIMPLIFIED = "simplified_login_i1"
```
I noticed the `i1` suffix on the variant in Firebase Console today, can you explain why you added it?

If the plan to prepare for another variant that follows site login for example, then I'd change the name to `simplified_login_wpcom` or something similar, my point is that it's not about versions, but about variants, and the name should be clear about it, which would be mapped to the enum as per my above remark 🙂.
Let me know what you think.

hafizrahman replied:
The idea is to match the design post naming: pecCkj-a6-p2

My thinking was: once the next iteration appears, we can use the same naming convention `simplified_login_i2`, `simplified_login_i3`, and so on, and it will make it easier to pair and refer to the related design posts.

At this point I am unsure if the next iteration is going to work on a completely different flow, or on the same flow but with more changes, so just following the design name feels easier.

hichamboushaba replied:
If we have a new design to the same variant, I don't believe it means having an additional variant, we will be updating the design of the said variant.
And if we have a new variant, then it will be different (not just new design).

I believe the `i1` suffix would just complicate things, variants are not for design iterations, but for different flows 🤔

hafizrahman replied:
Cool cool, let's go with `simplified_login_wpcom` then. I'll update in Firebase too.

hichamboushaba replied:
Thanks @hafizrahman 🙏

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/SimplifiedLoginExperiment.kt:
```diff
@@ -1,9 +1,42 @@
+        private const val VARIANT_SIMPLIFIED = "simplified_login_i1"
+    }
+
+    fun run() {
```
Should we rename this? something like `activate` would probably make more sense for our usage.

hafizrahman replied:
Good point! Thanks!

hafizrahman committed:
Update: Rename function to be clearer.

hafizrahman committed:
Update: Convert string to enum in repository, and rename enum to match Firebase values.

hafizrahman committed:
Merge branch 'trunk' into activate/simplified-login-experiment-setup

hafizrahman committed:
Update: variants naming to be "CONTROL" and "SIMPLIFIED_LOGIN_WPCOM"

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/SimplifiedLoginExperiment.kt:
```diff
@@ -1,12 +1,34 @@
-        STANDARD,
-        SIMPLIFIED
+        CONTROL,
+        SIMPLIFIED_LOGIN_WPCOM
```
This is just a comment to remind both of ourselves to make sure this PR is up-to-date with trunk before merging it, this is to avoid breaking `trunk`, as the `SIMPLIFIED` value is already used in some places.

hafizrahman replied:
Yup I forgot that but have added fix in https://github.com/woocommerce/woocommerce-android/pull/7623/commits/62d1491ce8b8d05241f251b569af31888cae5ab2

hafizrahman committed:
Add: log successful event for Simplified Login A/B test.

We consider a login success when "My Store" is loaded.

Also fixes small detekt issue.

hafizrahman commented:
Ready for another round of review @hichamboushaba 

I think the only addition is the placement of tracking inside MyStoreFragment.

hichamboushaba reviewed (approved):
Works well, nice work @hafizrahman 

I confirmed the AB test using the steps from the [official guide](https://firebase.google.com/docs/ab-testing/abtest-config#validate_your_experiment_on_a_test_device)  (with a small difference, I used FCM tokens instead of installation tokens)
